src/assets/AssetsDetectionController.ts

```diff
@@ -332,40 +332,47 @@ export class AssetsDetectionController extends BaseController<
         tokensToDetect.push(address);
       }
     }
+    const sliceOfTokensToDetect = [];
+    sliceOfTokensToDetect[0] = tokensToDetect.slice(0, 1000);
+    sliceOfTokensToDetect[1] = tokensToDetect.slice(
```

Comment:
Have you considered splitting it into 1000-length chunks, rather than two fixed chunks? For users with >2000 tokens I mean.

Reply:
The second chunk is not fixed length, it covers all the remaining tokens. Or should I split it into multiple chunks of 1000, but that might increase the number of external requests to check the balance.